packages/salesforcedx-vscode-core/src/commands/forceStopApexDebugLogging.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {
+  CliCommandExecutor,
+  Command,
+  CommandOutput,
+  SfdxCommandBuilder
+} from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import * as vscode from 'vscode';
+import { developerLogDebugLevels } from '.';
+import { nls } from '../messages';
+import { disposeTraceFlagExpiration } from '../traceflag-time-decorator';
+import {
+  CancelResponse,
+  ContinueResponse,
+  ParametersGatherer,
+  SfdxCommandlet,
+  SfdxCommandletExecutor,
+  SfdxWorkspaceChecker
+} from './commands';
+import {
+  debugLevelId,
+  prevApexCodeDebugLevel,
+  prevVFDebugLevel
+} from './forceStartApexDebugLogging';
+
+class ForceStopApexDebugLoggingExecutor extends SfdxCommandletExecutor<{}> {
+  public build(data: {}): Command {
+    return getRestoreLevelsCommand();
+  }
+
+  public execute(response: ContinueResponse<{}>): void {
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const cancellationToken = cancellationTokenSource.token;
+
+    const execution = new CliCommandExecutor(this.build(response.data), {
+      cwd: vscode.workspace.rootPath
+    }).execute(cancellationToken);
+
+    this.attachExecution(execution, cancellationTokenSource, cancellationToken);
+    execution.processExitSubject.subscribe(async data => {
+      if (data != undefined && data.toString() === '0') {
+        developerLogDebugLevels.turnOffLogging();
+        disposeTraceFlagExpiration();
+      }
+    });
+  }
+}
+
+export function restoreDebugLevels() {
+  if (prevApexCodeDebugLevel && prevVFDebugLevel) {
```

Comment:
Are these values updated in `forceStartApexDebugLogging`?

Reply:
These values should be pulled from the DeveloperLogDebugLevels object. Thanks for catching this